src/librustc_mir/transform/simplify_cfg.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use repr::*;
+use rustc::middle::const_eval::ConstVal;
+use std::mem;
+use transform::util;
+use transform::MIRPass;
+
+pub struct SimplifyCFG;
```

Comment:
Nit: Usually we provide a top-level `fn` like `simplify_cfg::simplify(...)` rather than calling a method on a singleton; not that I care very much. And yes the capitalization convention is `SimplifyCfg` (but I saw that mentioned elsewhere).